### PR TITLE
Use Node 6 to handle HTTP clientError scenarios correctly, fixes #567

### LINKED ISSUE #567: Connection closed without response on Heroku
Sometimes Heroku logs their H13 platform error "Connection closed without response".  This is reproducible by sending malformed HTTP requests to the service, and results from an interaction between Heroku router (which insists that the backend produce a response) and Node (which closes connections that send malformed requests without a response).  Neither of these behaviours are currently modifiable by us.

Node issue raised: https://github.com/nodejs/node/issues/4543
Heroku issue raised: https://help.heroku.com/tickets/320113

It seems like Heroku will be making a change to their router that will make the issue disappear, and Node contributors seem amenable to giving developers control over HTTP responses to malformed requests.  We should keep this issue open while the H13 errors are still occurring.

### PR DESCRIPTION
**Pull request by triblondon**
Before:

image

After:

image

### CODE REVIEW

File: bin/polyfill-service [hunk 1]
```diff
@@ -18,6 +18,10 @@ var startService = require('../service');
 var argv = minimist(process.argv.slice(2));
 var port = argv.port || Number(process.env.PORT) || 3000;
 
+process.on('uncaughtException', (err) => {
```
Comment (JakeChampion) [inline bin/polyfill-service:21]: What prompted the adding of this event handler?

Comment (triblondon) [inline bin/polyfill-service:21]: Initially I made a mistake in index.js and referred to the `sock` var as `socket`.  Until I added the code above, all I saw on the console was this:

```
3 May 12:38:09 - uncaughtException: db42aa429377452ebe9a3a5ad36fee73
```

Comment (JakeChampion) [inline bin/polyfill-service:21]: Should we send the error to Sentry as well?

Comment (triblondon) [inline bin/polyfill-service:21]: Seems sentry actually already logs it

https://app.getsentry.com/nextftcom/polyfill-service-dev/issues/126700489/

**Comment by JakeChampion** [comment]
LGTM 👍

**Pull request merged**